keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,1430 @@
+      value: "24"
+      effect: NoSchedule
+---
+# Batch training workload tolerates degraded devices
+kind: ResourceClaim
+metadata:
+  name: training-gpu-claim
+spec:
+  requests:
+  - name: batch-gpu
+    deviceClassName: device.example.com
+    tolerations:
+    # Accept devices with >1 hour error budget
+    - key: device.example.com/error-budget-in-hours
+      operator: Gt
+      value: "1"
+      effect: NoSchedule
+```
+
+#### Story 6 — Kubernetes version compatibility for critical workloads
+
+As a cluster operator managing a mixed-version Kubernetes cluster during rolling upgrades, I want to ensure critical workloads only run on nodes with Kubernetes version >= 1.20.0 due to specific API features they require, while allowing development workloads to tolerate older versions.
+
+Semantic version tolerations provide precise version-based placement without hardcoding specific versions in NodeAffinity rules.
+
+**Example Configuration:**
+
+```yaml
+# Older node running Kubernetes 1.19.5
+apiVersion: v1
+kind: Node
+metadata:
+  name: old-node-1
+spec:
+  taints:
+  - key: node.kubernetes.io/version
+    value: "1.19.5"
+    effect: NoSchedule
+---
+# Production workload requires Kubernetes >= 1.20.0
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: production-api
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node.kubernetes.io/version
+        operator: SemverGt
+        value: "1.20.0"
+        effect: NoSchedule
+      containers:
+      - name: api
+        image: myapp:latest
+---
+# Development workload tolerates older versions
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dev-workload
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node.kubernetes.io/version
+        operator: SemverGt
+        value: "1.33.0"  # Can run on 1.33+ nodes
+        effect: NoSchedule
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+**Value Format and Type Detection:**
+
+- **Automatic Type Detection**: The system automatically determines whether to use integer or semantic version comparison based on value format. Both toleration and taint values must be parseable as the same type for comparison to occur.
+
+- **Integer Comparison Requirements**:
+  - Values must be parseable as signed 64-bit integers
+  - Decimal numbers (e.g., `"95.5"`) are not supported to honor Kubernetes' avoidance of floating-point precision issues
+  - Leading zeros are preserved in parsing (`"0950"` vs `"950"` are numerically equal)
+
+- **Semantic Version Requirements**:
+  - Values must conform to [Semantic Versioning 2.0.0](https://semver.org/) specification requiring exactly 3 components (major.minor.patch)
+  - Supports both prefixed (`"v1.20.1"`) and non-prefixed (`"1.20.1"`) formats following Kubernetes conventions  
+  - Invalid semver strings (e.g., `"1.20"`, `"1.20.1.1"`, `"1"`) cause validation errors
+  - Pre-release versions (e.g., `1.20.0-alpha.1`) have lower precedence than release versions (`1.20.0`)
```

Comment:
Nit: the pre-release and build metadata points are part of semver, so we could drop them. Or if being super explicit is helpful, let's add the "per semver specification" suffix to the Pre-release versions line item as well.